### PR TITLE
Spec branch

### PR DESCRIPTION
**Pull request by EmiLoKiJu**
- Create a class called Solver. ✔️ 

- Create a method called factorial that takes one argument, an integer N, and returns the [factorial] (https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception. ✔️ 

- Create a method called reverse that takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it returns "olleh"). ✔️ 

- Create a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:

- When N is divisible by 3, return "fizz". ✔️ 
- When N is divisible by 5, return "buzz". ✔️ 
- When N is divisible by 3 and 5, return "fizzbuzz". ✔️ 

- Any other case, return N as a string (e.g. say N is 7 then return "7"). ✔️ 

- Your commit history should prove that you have used TDD. ✔️

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    end
+
+    it 'should return 1 when the number given is 0' do
+      expect(solver.factorial(0)).to eq(1)
+    end
+
+    it 'should return -1 when the number given is less than 0' do
+      expect(solver.factorial(-5)).to eq(-1)
+    end
+  end
```
Comment (abdulhamiid) [inline spec/solver_spec.rb]: According to the project requirements, The `#factorial` method should have tests that handle: * Negative value which raises an exception as an expectation 🤓

e.g `expect { resolver.factorial(-5) }.to raise_exception` 👍

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @EmiLoKiJu 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! :running_man: :runner: 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Write unit tests using Rspec.
- ✔️ Understand the TDD concept.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply spec/solver_spec.rb]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @EmiLoKiJu 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**